### ISSUE TITLE
Deno has panicked reading from stdin with the `--unstable-hmr` flag

### ISSUE THREAD

**Issue by oscarotero** [denoland/deno#21069]
I was testing the new HMR feature introduced in Deno 1.38 and noticed that the following code makes Deno to panick:

```
echo "import './main.js'" | deno run --unstable-hmr -
```

**Comment by dsherret** [comment]
Panics unwraping the clone of the watcher communicator:

https://github.com/denoland/deno/blob/76fbe85264eda4abf9b17f6ec915fdc6ea8c24b0/cli/factory.rs#L630

**Comment by linde12** [comment]
`run_from_stdin` uses `CliFactory::from_flags -> CliFactoryBuilder::build_from_flags` which does not receive a `watcher_communicator` (it will be `None`) whereas `run_script` will look at watch flags and call `run_with_watch` if they are present.

I guess we should raise an error if hmr is used without and file paths

**Comment by linde12** [comment]
I'd like to work on this one if possible. But do we want to silently ignore the flag (like what happens with --watch today) or do we want to explicitly error? I'm leaning towards explicitly erroring, but either is fine.

**Comment by oscarotero** [comment]
Why silence/show an error? I agree that it makes no sense to watch the code of the stdin, but it could watch the files imported by this code.

**Comment by linde12** [comment]
True, but the use cases for this seems very slim so it might not be worth the effort to implement. I don't know how large of a change that would be (would have to update file_watcher, run module, factory and probably more?), but again not sure if its worth? Need more input on this 👍

**Comment by oscarotero** [comment]
In my use case, I would like to test it with Lume, that [it's executed with this task](https://github.com/lumeland/lume.land/blob/main/deno.json#L18).

The reason to run the task in this way is because `deno run lume/cli.ts` doesn't work because `lume/` is not resolved with the import map.

This will be changed once https://github.com/denoland/deno/pull/20018 is landed.

**Comment by linde12** [comment]
Cool! Thanks for the update, i understand @oscarotero 

I have uptaded my PR to which enables `--unstable-hmr` and `--watch` to work with stdin. Waiting for some feedback before proceeding!
